Решение задач 6-module

AStruchenkov opened the pull request.

twangest committed:
Решение задачи 5-module 1-task

twangest committed:
Merge branch 'master' of https://github.com/js-tasks-ru/nodejs-20200512_astruchenkov

twangest committed:
Merge branch 'master' of https://github.com/js-tasks-ru/nodejs-20200512_astruchenkov

twangest committed:
Добавлено решение задач 6-module

twangest committed:
Добавлена исправленная версия решения 6-module

twangest committed:
исправляем зависимости для travis

twangest committed:
Merge branch 'master' of https://github.com/AStruchenkov/nodejs-20200512_astruchenkov

twangest committed:
Исправлены зависимости для travis

twangest committed:
Исправления для pull-request

twangest committed:
файл теста 6-module 3-task вернул к исходному виду

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

```diff
Модели товара и категории
+ у модели есть поля title и subcategories
+ поле title имеет правильную конфигурацию
+ поле subcategories имеет правильную конфигурацию
+ у модели есть поля: title, description, price, category, subcategory и images
+ поле title имее�� правильную конфигурацию
+ поле description имеет правильную конфигурацию
+ поле price имеет правильную конфигурацию
+ поле category имеет правильную конфигурацию
+ поле subcategory имеет правильную конфигурацию
+ поле images имеет правильную конфигурацию
```
```diff
Онлайн-магазин (товары и категории)
+ по запросу должен возвращаться список категорий
+ если товары есть в базе - должен вернуться массив с товарами
+ если товаров не найдено - должен возвр��щаться пустой массив
+ товары по существующей подкатегории
+ сервер должен вернуть ошибку если идентификатор невалидный
+ сервер должен вернуть статус 404
+ сервер должен вернуть товар по его айди
```
```diff
Поиск товаров
+ у модели есть текстовый индекс с именем TextSearchIndex
+ индекс должен включа��ь поля title и description
+ индекс содержит указание весов для полей
+ язык по умолчанию - русский
+ товары должны возвращаться в порядке релевантности
+ если ничего не найдено - пустой массив
```
```diff
Настройка локальной стратегии
+ поле usernameField должно должно содержать email
- стратегия должна возвращать ошибку если указан несуществующий email
- стратегия должна возвращать ошибку если указан неверный пароль
- стратегия должна возвращать объект пользователя если email и пароль верные
```

twangest committed:
Добавлен 7-module 1-task

Merge branch 'master' of https://github.com/js-tasks-ru/nodejs-20200512_astruchenkov

twangest committed:
Добавлено решение задачаи 7-module 1-task

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Модели товара и категории
+ у модели есть поля title и subcategories
+ поле title имеет правильную конфигурацию
+ поле subcategories имеет правильную конфигурацию
+ у модели есть поля: title, description, price, category, subcategory и images
+ поле title имее�� правильную конфигурацию
+ поле description имеет правильную конфигурацию
+ поле price имеет правильную конфигурацию
+ поле category имеет правильную конфигурацию
+ поле subcategory имеет правильную конфигурацию
+ поле images имеет правильную конфигурацию
```
```diff
Онлайн-магазин (товары и категории)
+ по запросу должен возвращаться список категорий
+ если товары есть в базе - должен вернуться массив с товарами
+ если товаров не найдено - должен возвр��щаться пустой массив
+ товары по существующей подкатегории
+ сервер должен вернуть ошибку если идентификатор невалидный
+ сервер должен вернуть статус 404
+ сервер должен вернуть товар по его айди
```
```diff
Поиск товаров
+ у модели есть текстовый индекс с именем TextSearchIndex
+ индекс должен включа��ь поля title и description
+ индекс содержит указание весов для полей
+ язык по умолчанию - русский
+ товары должны возвращаться в порядке релевантности
+ если ничего не найдено - пустой массив
```
```diff
Настройка локальной стратегии
+ поле usernameField должно должно содержать email
+ стратегия должна возвращать ошибку если указан несуществующий email
+ стратегия должна возвращать ошибку если указан неверный пароль
+ стратегия должна возвращать объект пользователя если email и пароль верные
```
